server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/EsMonitor.java

```diff
@@ -96,8 +109,8 @@ public int getNumberOfNodes() {
     for (Map.Entry<String, NodeStats> entry : nodesStats.getNodesMap().entrySet()) {
 
       LinkedHashMap<String, Object> nodeAttributes = new LinkedHashMap<>();
-      nodes.put(entry.getKey(), nodeAttributes);
       NodeStats stats = entry.getValue();
+      nodes.put(stats.getNode().getName(), nodeAttributes);
```

Comment:
`stats.getNode()` should be stored in a variable to have consistent values and avoid potential perf overhead